src/main/java/com/spotify/docker/AbstractDockerMojo.java

```diff
@@ -59,6 +72,22 @@ public void execute() throws MojoExecutionException {
         builder.uri(dockerHost);
       }
 
+      final AuthConfig.Builder authConfigBuilder = AuthConfig.builder();
+      if (!isNullOrEmpty(dockerRegistryUsername)) {
+        authConfigBuilder.username(dockerRegistryUsername);
+      }
+      if (!isNullOrEmpty(dockerRegistryEmail)) {
+        authConfigBuilder.email(dockerRegistryEmail);
+      }
+      if (!isNullOrEmpty(dockerRegistryPassword)) {
+        authConfigBuilder.password(dockerRegistryPassword);
+      }
+      if (!isNullOrEmpty(dockerRegistryServerAddress)) {
+        authConfigBuilder.serverAddress(dockerRegistryServerAddress);
+      }
+
+      builder.authConfig(authConfigBuilder.build());
```

Comment:
I'd say some combinations here don't make sense, like password without username, for example. Should it complain when such a combination is detected?